File: Regular Expressions/Code/chapter2/2-3/recipe2-3.pl
#!/usr/bin/perl -w
use strict;

my $mystr = $ARGV[0] || die "Please supply a parameter";

if ( $mystr =~ s/^https?\:\/\/([^\/]+)\/?.*$/$1/ )
{
	print "The domain is '" . $mystr . "'\n";
}
else
{
	print "No hostname or anything like that in this string.\n";
}

File: Regular Expressions/Code/chapter2/2-3/recipe2-3.py
#!/usr/bin/python
import re
import sys

nargs = len(sys.argv)

if nargs > 1:

	mystr = sys.argv[1]

	r = re.compile( r'^https?:\/\/([^\/]+)\/?.*$', re.I )

	hostname = r.sub( r'\1', mystr )

	print 'Hostname is: ' + hostname,

else:
	print 'Filename?  Anyone?  Anyone?',
